Documentation/gettingstarted/kubeproxy-free.rst

```diff
@@ -105,31 +108,179 @@ is ready to operate:
+        --set global.k8sServiceHost=API_SERVER_IP \\
+        --set global.k8sServicePort=API_SERVER_PORT
 
 If ``kube-apiserver`` was configured to use a non-default NodePort port range,
 then the same range must be passed to Cilium via the ``global.nodePort.range``
 option.
 
+kube-proxy Replacement Modes
+############################
+
+Cilium's BPF kube-proxy replacement can be configured in several modes, i.e. it can
+replace kube-proxy entirely or it can co-exist with kube-proxy on the system if the
+underlying Linux kernel requirements do not support a full kube-proxy replacement.
+
+This section therefore elaborates on the various ``global.kubeProxyReplacement`` options:
+
+- ``global.kubeProxyReplacement=strict``: This option expects a kube-proxy-free
+  Kubernetes setup where Cilium is expected to fully replace all kube-proxy
+  functionality. Once the Cilium agent is up and running, it takes care of handling
+  Kubernetes services of type ClusterIP, NodePort, ExternalIPs and LoadBalancer.
+  If the underlying kernel version requirements are not met (see :ref:`kubeproxy-free`
+  note), then the Cilium agent will bail out on start-up with an error message.
+
+- ``global.kubeProxyReplacement=probe``: This option is intended for a hybrid setup,
+  that is, kube-proxy is running in the Kubernetes cluster where Cilium partially
+  replaces and optimizes kube-proxy functionality. Once the Cilium agent is up and
+  running, it probes the underlying kernel for the availability of needed BPF kernel
+  features and, if not present, disables a subset of the functionality in BPF by
+  relying on kube-proxy to complement the remaining Kubernetes service handling. The
+  Cilium agent will emit an info message into its log in such case. For example, if
+  the kernel does not support :ref:`host-services`, then the ClusterIP translation
+  for the node's host-namespace is done through kube-proxy's iptables rules.
+
+- ``global.kubeProxyReplacement=partial``: Similarly to ``probe``, this option is
+  intended for a hybrid setup, that is, kube-proxy is running in the Kubernetes cluster
+  where Cilium partially replaces and optimizes kube-proxy functionality. As opposed to
+  ``probe`` which checks the underlying kernel for available BPF features and automatically
+  disables components responsible for the BPF kube-proxy replacement when kernel support
+  is missing, the ``partial`` option requires the user to manually specify which components
+  for the BPF kube-proxy replacement should be used. Similarly to ``strict`` mode, the
+  Cilium agent will bail out on start-up with an error message if the underlying kernel
+  requirements are not met. For fine-grained configuration, ``global.hostServices.enabled``,
+  ``global.nodePort.enabled`` and ``global.externalIPs.enabled`` can be set to ``true``.
+  By default all three options are set to ``false``. A few example configurations for the
+  ``partial`` option are provided below.
```

Comment:
Can someone remind me why users would care to enable or disable individual features of kube-proxy-free?

I can see how this might be useful for development, or maybe in a tricky case in prod where you really just need to turn off some functionality. Just wondering whether we should even document or encourage this option.